content/systems/filecoin_blockchain/storage_power_consensus/_index.md

```diff
@@ -81,15 +81,12 @@ included in a Filecoin block. This is shown below.
 ### Get drand randomness for VM
 
 For operations such as PoRep creation, proof validations, or anything that
-requires randomness for the Filecoin VM, the following method shows how to
-extract the drand entry from the chain.
+requires randomness for the Filecoin VM, there should be a method that
+extracts the drand entry from the chain correctly.
 Note that the round may span multiple filecoin epochs if drand is slower; the
 lowest epoch number block will contain the requested beacon entry. Similarly, if
 there has been null rounds where the beacon should have been inserted, we need
-to iterate on the chain to find where the entry is inserted.
-
-
-{{<embed src="https://github.com/filecoin-project/lotus/blob/master/chain/vm/runtime.go" lang="go" symbol="GetRandomnessFromBeacon">}}
+to iterate on the chain to find where the entry is inserted. In addition, the next non-null block must contain the drand entry requested by definition.
```

Comment:
Nice - I would simply replace "In addition" by "Specifically", to say that it _is_ the way of fetching the randomness from chain, not an "additional property".